etcd-manager/pkg/volumes/scaleway/volumes.go

```diff
@@ -0,0 +1,248 @@
+	}
+
+	var localEtcdVolumes []*volumes.Volume
+	for _, volume := range allEtcdVolumes {
+		// Only volumes from the same location can be mounted
+		if volume.Zone == "" {
+			klog.Warningf("failed to find volume location for %s(%d)", volume.Name, volume.ID)
+			continue
+		}
+		volumeLocation := volume.Zone
+		if volumeLocation != a.zone {
+			continue
+		}
+		klog.V(2).Infof("Found attachable volume %s(%d) with status %q", volume.Name, volume.ID, volume.State)
+
+		localEtcdVolume := &volumes.Volume{
+			ProviderID: volume.ID,
+			Info: volumes.VolumeInfo{
+				Description: a.clusterName + "-" + volume.ID,
+			},
+			MountName: "scw-" + volume.ID,
+			EtcdName:  "vol-" + volume.ID,
+		}
+
+		if volume.Server != nil {
+			localEtcdVolume.AttachedTo = volume.Server.ID
+			if volume.Server.ID == a.server.ID {
+				localEtcdVolume.LocalDevice = fmt.Sprintf("%s%s", localDevicePrefix, volume.ID)
+			}
+		}
+
+		localEtcdVolumes = append(localEtcdVolumes, localEtcdVolume)
+	}
+
+	return localEtcdVolumes, nil
+}
+
+// FindMountedVolume returns the device where the volume is mounted to the running server.
+func (a *ScwVolumes) FindMountedVolume(volume *volumes.Volume) (string, error) {
+	device := volume.LocalDevice
+
+	klog.V(2).Infof("Finding mounted volume %q", device)
+	_, err := os.Stat(volumes.PathFor(device))
+	if err == nil {
+		klog.V(2).Infof("Found mounted volume %q", device)
+		return device, nil
+	}
+
+	if !os.IsNotExist(err) {
+		return "", fmt.Errorf("failed to find local device %q: %v", device, err)
+	}
+
+	// When not found, the interface says to return ("", nil)
+	return "", nil
+}
+
+// AttachVolume attaches the specified volume to the running server and returns the mountpoint if successful.
+func (a *ScwVolumes) AttachVolume(volume *volumes.Volume) error {
+	for {
+		volumeResp, err := a.instanceAPI.GetVolume(&instance.GetVolumeRequest{
+			VolumeID: volume.ProviderID,
+			Zone:     a.zone,
+		})
+		if err != nil || volumeResp.Volume == nil {
+			return fmt.Errorf("failed to get info for volume id %q: %w", volume.ProviderID, err)
+		}
+
+		scwVolume := volumeResp.Volume
+		if scwVolume.Server != nil {
+			if scwVolume.Server.ID != a.server.ID {
+				return fmt.Errorf("found volume %s(%s) attached to a different server: %s", scwVolume.Name, scwVolume.ID, scwVolume.Server.ID)
+			}
+			klog.V(2).Infof("Attached volume %s(%d) to the running server", scwVolume.Name, scwVolume.ID)
+			volume.LocalDevice = fmt.Sprintf("%s%s", localDevicePrefix, scwVolume.ID)
+			return nil
+		}
+
+		//err = a.instanceAPI.ServerActionAndWait(&instance.ServerActionAndWaitRequest{
+		//	ServerID: a.server.ID,
+		//	Zone:     a.zone,
+		//	Action:   instance.ServerActionPoweroff,
+		//})
+		//if err != nil {
+		//	return fmt.Errorf("failed to power-off server %s before attaching volume", a.server.ID)
+		//}
+
+		klog.V(2).Infof("Attaching volume %s(%d) to the running server", scwVolume.Name, scwVolume.ID)
```

Comment:
Right now let's support only block volume that can be added without restarting a server. That seems easier to handle for a first version rather than having local volume also supported.